### ISSUE TITLE
Pimgento-2-API Fails everytime - 504 Error - 6mins exactly

### ISSUE THREAD

**Issue by gconstable** [Agence-DnD/PIMGento-2-API#80]
# Symptom
- Importing products via api fails with no error reported via pimgento (spinner just spins).
- Google chrome browser (dev tools) reports a 504 error.
- Please see screenshot
![screenshot at 2019-02-13 15-11-06](https://user-images.githubusercontent.com/40803397/52721839-6cea1b80-2fa2-11e9-86c7-77a7ed5bd213.png)

# Simple Diagnosis
- After repeating the process approx 12 times i can see via google chrome that if a query lasts longer than 6 mins then the api request dies.

# Actions taken so far with no luck
- Increased ram on VM to 50+ GB thinking easticsearch on akeno was a problem.
- Changed credentials on api connection (didnt think this work).
- Increased session.gc_maxlifetime within php on pim server to 3600.
- Increased sslsession cache timer to 600 within apache2.
- Increased magento2 pimgento pagination size to 100.
- Increased hdd size on VM by 25%.

# In advance
I would like to say thanks in advance for taking the time to look at this.

**Comment by Dnd-Gimix** [comment]
Hello @gconstable ,

Thank you for using PIMGento 2 API.
It seems to be a server timeout that is set to 360s and that blocks your request.

Do you run Magento on NGINX or APACHE ? Do you know the request timeout value ?

Also, what is your PHP max_execution_time ?
Thank you.

Regards,

**Comment by gconstable** [comment]
Hi and thanks for getting back to me.

The server is running Apache.

I checked PHP configuration and it's set too 3600 seconds.  It was
originally 360 but I changed this days ago and reboot the server with no
luck.

I also looked for SQL timers, .htaccess files with no luck.

I am unsure though if an API URL is treated differently or not by the
server in terms of timers.  I could not locate where these settings could
be.

Any other thoughts?

**Comment by gconstable** [comment]
I also temp enabled phpinfo if you would like to see.

https://35.204.185.170/phpinfo.php

**Comment by Dnd-Gimix** [comment]
@gconstable ,

In you php_info(), PHP max_execution_time seems to be set at 120. Maybe you want a greater value for this configuration ?
Also, could you increase the "Timeout" parameter of apache to see if it fix your issue ?

Regards,

**Comment by gconstable** [comment]
Hi,
  Many thanks for getting back to me.  I went through all the settings again on the pim server on both apache and php with no luck.  However, I did manage to find the issue.  The issue was the php timers set on the Magento store.  I had to extend the php max_execution time and then after that I had to extend the admin session time within magento it’s self as the shop account was terminating before the script would finish.

Many thanks again

Best Regards, Gary Constable
